app/com/arpnetworking/kairos/client/KairosDbClient.java

```diff
@@ -15,57 +15,79 @@
 
+import java.time.Duration;
+import java.time.Instant;
 import java.util.concurrent.CompletionStage;
 
 /**
  * Client for accessing KairosDB APIs.
  *
  * @author Brandon Arp (brandon dot arp at smartsheet dot com)
  */
-public interface KairosDbClient {
+public abstract class KairosDbClient {
     /**
      * Executes a query for datapoints from  KairosDB.
      *
      * @param query the query
      * @return the response
      */
-    CompletionStage<MetricsQueryResponse> queryMetrics(MetricsQuery query);
+    public abstract CompletionStage<MetricsQueryResponse> queryMetrics(MetricsQuery query);
 
     /**
      * Queries KairosDB for metric names.
      *
      * @return the response
      */
-    CompletionStage<MetricNamesResponse> queryMetricNames();
+    public abstract CompletionStage<MetricNamesResponse> queryMetricNames();
 
     /**
      * Queries KairosDB for a list of tags associated with a metric.
      *
      * @param query simplified metrics query to retrieve tags for
      * @return a query response only containing tags with values for each metric
      */
-    CompletionStage<MetricsQueryResponse> queryMetricTags(TagsQuery query);
+    public abstract CompletionStage<MetricsQueryResponse> queryMetricTags(TagsQuery query);
 
     /**
      * Query tag names in KairosDb.
      *
      * @return the response
      */
-    CompletionStage<TagNamesResponse> listTagNames();
+    public abstract CompletionStage<TagNamesResponse> listTagNames();
 
     /**
      * Persist metric data points.
      *
      * @param metricDataPoints the metric data points to persist
      * @return completion stage indicating success or failure of operation
      */
-    CompletionStage<Void> addDataPoints(ImmutableList<MetricDataPoints> metricDataPoints);
+    public abstract CompletionStage<Void> addDataPoints(ImmutableList<MetricDataPoints> metricDataPoints);
+
+    /**
+     * Wrapper around {@link #queryMetricTags(TagsQuery)}, to get the set of tag-values for a single metric.
+     *
+     * @param metricName the name of the metric to get the tag names/values for.
+     * @param time the time to get the tags at.
+     * @return the set of tag names/values that the given metric has at <i>roughly</i> the given time.
+     *   (Due to KairosDB implementation issues, it's actually the set of all names/values over a couple-week period
+     *    that includes the given time.)
+     */
+    public CompletionStage<ImmutableMultimap<String, String>> queryMetricTags(final String metricName, final Instant time) {
+        return queryMetricTags(ThreadLocalBuilder.build(TagsQuery.Builder.class, tqb -> tqb
+                .setMetrics(ImmutableList.of(ThreadLocalBuilder.build(MetricTags.Builder.class, mtb -> mtb.setName(metricName))))
+                .setStartTime(time)
+                .setEndTime(time.plus(Duration.ofDays(1)))
+        )).thenApply(response -> response.getQueries().get(0).getResults().get(0).getTags());
+    }
```

Comment:
I think that this method should also be used by the RollupGenerator, but its current structure makes that hard to do. I might perform the necessary re-factor in a separate PR.